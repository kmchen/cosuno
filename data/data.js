export const data =
{
   "company":[
      {
         "name":"AUTO1 Group",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Blacklane",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"Blinkist",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Bonify",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"Camunda",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Cologne"
      },
      {
         "name":"ChartMogul",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
      {
         "name":"Contentful",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["Software"],
         "city": "Cologne"
      },
      {
         "name":"Cosuno",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["construction", "software"],
         "city": "Berlin"
      },
      {
         "name":"Coya",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Crate.io",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Dalia Research",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
      {
         "name":"door2door",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"Eyeo",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Cologne"
      },
      {
         "name":"FreightHub",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"German Autolabs",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"GetYourGuide",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Dusseldorf"
      },
      {
         "name":"GoEuro",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"Helping",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Infarm",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Junique",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"MoBerries",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Morressier",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Movinga",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"N26",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"PEAT",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"PPRO",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"Quentic",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Cologne"
      },
      {
         "name":"ResearchGate",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Skoove",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"SnapShot",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
      {
         "name":"Watchmaster",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Wundertax",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"Zeotap",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
   ]
};
