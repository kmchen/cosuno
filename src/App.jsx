import React from 'react';
import { connect } from 'react-redux';
import Fuse from 'fuse.js';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import SearchBar from './components/SearchBar';
import Footer from './components/Footer';
import NavBar from './components/NavBar';
import Results from './components/Results';
import PUSH_SEARCH_RESULT from './actions/actions';

const options = {
  shouldSort: true,
  threshold: 0.0,
  location: 0,
  distance: 100,
  maxPatternLength: 32,
  minMatchCharLength: 1,
  keys: ['name'],
};

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
  },
}));

export function AppComponent({ dispatch, data, selectedCompany, fuse }) {
  const classes = useStyles();
  const dispatchResult = result => {
    const company = data.filter(val => val.name === result).shift();
    dispatch({ type: PUSH_SEARCH_RESULT, payload: [company] });
  };
  return (
    <div className={classes.root}>
      <CssBaseline />
      <NavBar />
      <main>
        <SearchBar setResult={dispatchResult} fuse={fuse} />
        <Results selectedCompany={selectedCompany} />
      </main>
      <Footer />
    </div>
  );
}

/* eslint-disable react/forbid-prop-types */
AppComponent.propTypes = {
  data: PropTypes.array.isRequired,
  selectedCompany: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
  fuse: PropTypes.object.isRequired,
};

const mapStateToProps = state => {
  const { data, selectedCompany } = state;
  return { data, selectedCompany, fuse: new Fuse(data, options) };
};

export const App = connect(
  mapStateToProps,
  null,
)(AppComponent);
