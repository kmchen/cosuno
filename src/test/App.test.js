import React from 'react';
import { shallow, configure } from 'enzyme';
import Fuse from 'fuse.js';
import Adapter from 'enzyme-adapter-react-16';
import { AppComponent } from '../App';
import { data } from '../../data/data.js';

configure({ adapter: new Adapter() });

const options = {
  shouldSort: true,
  threshold: 0.0,
  location: 0,
  distance: 100,
  maxPatternLength: 32,
  minMatchCharLength: 1,
  keys: ['name'],
};

describe('App component', () => {
  let appComponent;

  beforeEach(() => {
    const props = {
      data: data.company,
      fuse: new Fuse(data, options),
      selectedCompany: [],
      dispatch: jest.fn(),
    };
    appComponent = shallow(<AppComponent {...props} />);
  });

  it('should render App component', () => {
    expect(appComponent).toMatchSnapshot();
  });
});
