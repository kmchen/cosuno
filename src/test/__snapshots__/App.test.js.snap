// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
<div
  className="makeStyles-root-1"
>
  <CssBaseline />
  <NavBar />
  <main>
    <SearchBar
      fuse={
        e {
          "list": Object {
            "company": Array [
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Hamburg",
                "name": "AUTO1 Group",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Frankfurt",
                "name": "Blacklane",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Blinkist",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Berlin",
                "name": "Bonify",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Cologne",
                "name": "Camunda",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "ChartMogul",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Cologne",
                "name": "Contentful",
                "specialty": Array [
                  "Software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Berlin",
                "name": "Cosuno",
                "specialty": Array [
                  "construction",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Hamburg",
                "name": "Coya",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Crate.io",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Dalia Research",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Frankfurt",
                "name": "door2door",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Cologne",
                "name": "Eyeo",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Frankfurt",
                "name": "FreightHub",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Frankfurt",
                "name": "German Autolabs",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Dusseldorf",
                "name": "GetYourGuide",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Frankfurt",
                "name": "GoEuro",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Hamburg",
                "name": "Helping",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Infarm",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Hamburg",
                "name": "Junique",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "MoBerries",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Morressier",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Frankfurt",
                "name": "Movinga",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Berlin",
                "name": "N26",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "PEAT",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Berlin",
                "name": "PPRO",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Cologne",
                "name": "Quentic",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "ResearchGate",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Skoove",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "SnapShot",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Hamburg",
                "name": "Watchmaster",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Berlin",
                "name": "Wundertax",
                "specialty": Array [
                  "fintech",
                  "software",
                ],
              },
              Object {
                "Logo": "https://source.unsplash.com/random",
                "city": "Munich",
                "name": "Zeotap",
                "specialty": Array [
                  "ecommerce",
                  "software",
                ],
              },
            ],
          },
          "options": Object {
            "distance": 100,
            "findAllMatches": false,
            "getFn": [Function],
            "id": null,
            "includeMatches": false,
            "includeScore": false,
            "isCaseSensitive": false,
            "keys": Array [
              "name",
            ],
            "location": 0,
            "matchAllTokens": false,
            "maxPatternLength": 32,
            "minMatchCharLength": 1,
            "shouldSort": true,
            "sortFn": [Function],
            "threshold": 0,
            "tokenSeparator": / \\+/g,
            "tokenize": false,
            "verbose": false,
          },
        }
      }
      setResult={[Function]}
    />
    <Results
      selectedCompany={Array []}
    />
  </main>
  <Footer />
</div>
`;
